proposals/5/README.md

```diff
@@ -0,0 +1,103 @@
+****---
+title: Decomposition of Jenkins X
+linktitle: Decomposition of Jenkins X
+description: Decomposition of Jenkins X
+type: docs
+weight: 50
+---
+
+## 1. Overview
+
+This document outlines the work for decomposing Jenkins X into a series of reusable libraries.
+
+## 1.1 Motivation
+
+At the moment, the source code that comprises the main JX binary is tightly coupled and difficult to work on.  Changes have
+unintended consequences and this makes the codebase fragile.
+
+One of the design goals in Accelerate is to have a **Loosely Coupled Architecture** - in particular: 
+
+_The goal is for your architecture to support the ability of teams to get their work done—from design through to deployment—without requiring high-bandwidth communication between teams._
+
+## 1.2 Background
+
+The main [JX repository](https://github.com/jenkins-x/jx) contains over 250k lines of code, creating Apps / Extensions / Binary Plugins is difficult without adding a dependency back onto the JX repo.
+
+It should be possible to separate the repository out into a series of reusable libraries so only what is needed can be imported.
+
+## 2. Design
+
+There are currently a number of proposed repositories/modules, some of these are currently in flight.
+
+### Existing Modules
+
+* [jx-api](https://github.com/jenkins-x/jx-api): JX CRDs / Installation Requirements
+* [jx-kube-client](https://github.com/jenkins-x/jx-kube-client): A helper module to create a Kubernetes rest config
+* [jx-vault-client](https://github.com/jenkins-x/jx-vault-client): longer term, it may make sense to deprecate this.
+* [go-scm](https://github.com/jenkins-x/go-scm): All 
+
+### Proposed Modules
+
+* jx-kube: It may make sense to refactor some of the higher level utils/kube functions within JX into its own package to make them reusable.
+* Version Streams: Would these make sense to be moved out of JX
+* Gitter?
+* Updatebot (jx create pr)? This would have value outside of the core JX codebase 
+* Storage APIs?
+* Cloud APIs?
+
+## 2.1. Action Plan
+
+1. *Agree on a way forward!*
+1. Ensure [Kubernetes Dependency Updates PR](https://github.com/jenkins-x/jx/pull/7313) is merged.
+1. Re-introduce [jx-api](https://github.com/jenkins-x/jx-api) as a dependency within [jx](https://github.com/jenkins-x/jx)
```

Comment:
I'd throw something in here mentioning the implications for downstream - like, I should probably move Lighthouse to `jx-api` before this, right?

Reply:
Eventually, I'd like to get there, but it'll be a while - Lighthouse does need to track `PipelineActivity`s and kick off metapipelines, so until/unless that behavior gets moved out of Lighthouse proper (either into different controllers in the Lighthouse repo or into new controllers in jx), Lighthouse does need to depend on jx.